### ISSUE TITLE
Can't decltype outer class' member if inner has a templated base

### ISSUE THREAD

**Issue by Alcaro** [llvm/llvm-project#55812]
```c++
template<typename T> class base {};

struct outer {
	int member;
	
	template<typename T>
	struct inner : base<T>
	{
		void fn()
		{
			decltype(outer::member) var1;
			decltype(&outer::member) var2;
			decltype(outer().member) var3;
		}
	};
};

template void outer::inner<int>::fn();
```
-> var1 throws an error: 'outer::member' is not a member of class 'outer::inner<int>'

https://godbolt.org/z/5Kxhq5YfG

I admittedly don't know if the above code is valid C++, but considering var2 and var3 work, and it also compiles if I change base\<T\> to base\<int\>, this is, at minimum, a misleading error message.

e: no, bad github markdown, stop eating my \<T\>

**Comment by Alcaro** [comment]
Fixed on trunk, closing (still present on 16)
